src/Randomizer.SMZ3.Tracking/AutoTracking/USB2SNESConnector.cs

```diff
@@ -0,0 +1,178 @@
+
+                Thread.Sleep(1000);
+
+                _logger.LogInformation($"Requesting device list");
+
+                Task.Run(() => _client.Send(JsonSerializer.Serialize(new USB2SNESRequest()
+                {
+                    Opcode = "DeviceList",
+                    Space = "SNES"
+                })));
+            });
+
+            _client.DisconnectionHappened.Subscribe(info =>
+            {
+                _lastMessage = null;
+                _logger.LogError(info.Exception, $"USB2SNES connection lost {info.Type}");
+                if (_isConnected)
+                {
+                    _isConnected = false;
+                    OnDisconnected?.Invoke(this, new());
+                }
+            });
+
+            _client.MessageReceived.Subscribe(msg => HandleUSB2SNESResponse(msg));
+            _client.Start();
+        }
+
+        /// <summary>
+        /// When a connection with the emulator is established
+        /// </summary>
+        public event EventHandler? OnConnected;
+
+        /// <summary>
+        /// When the connection with the emulator has been lost
+        /// </summary>
+        public event EventHandler? OnDisconnected;
+
+        /// <summary>
+        /// When a message from the emulator with memory has been returned
+        /// </summary>
+        public event EmulatorDataReceivedEventHandler? MessageReceived;
+
+        /// <summary>
+        /// If the connector is currently connected to the emulator
+        /// </summary>
+        /// <returns></returns>
+        public bool IsConnected() => _isConnected;
+
+        /// <summary>
+        /// Sends a message to the emulator
+        /// </summary>
+        /// <param name="message">The message to send to the emulator</param>
+        public void SendMessage(EmulatorAction message)
+        {
+            if (_lastMessage != null) return;
+            _lastMessage = message;
+            var convertedAddress = message.Address + (message.Domain == MemoryDomain.CartRAM ? 0x700000 : 0x770000);
+            var address = convertedAddress.ToString("X");
+            var length = message.Length.ToString("X");
+
+            if (message.Type == EmulatorActionType.ReadBlock)
+            {
+                _client.Send(JsonSerializer.Serialize(new USB2SNESRequest()
+                {
+                    Opcode = "GetAddress",
+                    Space = "SNES",
+                    Operands = new List<string>() { address, length }
+                }));
+            }
+            
+        }
+
+        /// <summary>
+        /// Closes the connection so that the connector can be destroyed
+        /// </summary>
+        public void Dispose()
+        {
+            try
+            {
+                _client.Dispose();
+            }
+            catch (ObjectDisposedException e)
+            {
+                _logger.LogError(e, "Already disposed USB2SNES client");
```

Comment:
You don't need to log this. This is kind of like getting a "File not found" error when you're trying to delete a file — the intended outcome was achieved either way.